src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,98 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[DEFAULT_CAPACITY];
+        values = (V[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putNullKey(value);
+        } else {
+            int index = indexOf(key);
+            if (index != -1) {
+                values[index] = value;
+            } else {
+                ensureCapacity();
+                keys[size] = key;
+                values[size] = value;
+                size++;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        if (key == null) {
+            return getNullKey();
+        }
+        int index = indexOf(key);
+        return (index != -1) ? (V) values[index] : null;
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    private int indexOf(K key) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] != null && keys[i].equals(key)) {
+                return i;
+            }
+        }
+        return -1;
```

Comment:
Please combine condition to work with both null key and nonNull key
With this approach you will not need to duplicate and handle null and nonNull cases separately
```suggestion
    private int indexOf(K key) {
        for (int i = 0; i < size; i++) {
            if (keys[i] == key || keys[i] != null && keys[i].equals(key)) {
                return i;
            }
        }
        return -1;
```